docs/topics/OAuth2.md

```diff
@@ -56,68 +41,229 @@ Scopes provide access to certain resources of a user's account. Your API client
+- `client_secret` - your application's client secret
+- `grant_type` - must be set to `refresh_token`
+- `refresh_token` - the user's refresh token
+- `redirect_uri` - your `redirect_uri`
+
+###### Refresh Token Exchange Example
+
+```python
+def refresh_token(refresh_token):
+  data = {
+    'client_id': '332269999912132097',
+    'client_secret': '456265548123452097',
+    'grant_type': 'refresh_token',
+    'refresh_token': refresh_token,
+    'redirect_uri': 'https://nicememe.website'
+  }
+  headers = {
+    'Content-Type': 'application/x-www-form-urlencoded'
+  }
+  r = requests.post('https://discordapp.com/api/oauth2/token', data, headers)
+  r.raise_for_status()
+  return r.json()
+```
+
+Boom; fresh [access token response](#DOCS_OAUTH2/authorization-code-grant-access-token-response)!
+
+## Implicit Grant
+
+The implicit OAuth2 grant is a simplified flow optimized for in-browser clients. Instead of issuing the client an authorization code to be exchanged for an access token, the client is directly issued an access token. The URL is formatted as follows:
+
+###### Implicit URL Example
 
 ```
-https://discordapp.com/api/oauth2/authorize?client_id=157730590492196864&scope=bot&permissions=0&redirect_uri=https%3A%2F%2Fnicememe.website
+https://discordapp.com/oauth2/authorize?response_type=token&client_id=290926444748734499&state=15773059ghq9183habn&scope=identify
 ```
 
-`client_id` is your _bot's_ application ID. `permissions` is an integer following the [permissions](#DOCS_PERMISSIONS/bitwise-permission-flags) format. If supplied, the `redirect_uri` will include additional query string parameters on redirect: `guild_id`, the guild the _bot_ has joined, and `permissions`, the same integer sent at the start of the flow.
+On redirect, your redirect URI will contain additional **URI fragments**: `access_token`, `token_type`, `expires_in`, and `scope`. **These are not querystring parameters.** Be mindful of the "#" character:
 
-### Adding Webhooks to Channels
+###### Implicit Redirect URL Example
 
-A URL can be generated that redirects authenticated users to the add-webhook flow, by using the following format (this utilizes the OAuth2 authentication authorization code flow, which requires a server-side application):
+```
+https://findingfakeurlsisprettyhard.tv/#access_token=RTfP0OK99U3kbRtHOoKLmJbOn45PjL&token_type=Bearer&expires_in=604800&scope=identify&state=15773059ghq9183habn
+```
+
+There are tradeoffs in using the implicit grant flow. It is both quicker and easier to implement, but rather than exchanging a code and getting a token returned in a secure HTTP body, the access token is returned in the URI fragment, which makes it possibly exposed to unauthorized parties. If you choose to implement the implicit grant, we highly recommend using the `state` parameter to help protect against cross-site request forgery. You also are not returned a refresh token, so the user must explicitly re-authorize once their token expires.
```

Comment:
> If you choose to implement the implicit grant, we highly recommend using the `state` parameter to help protect against cross-site request forgery.

is general guidance for all, not just this grant, so maybe can just drop this.

also, this recommendation would be better in the authorization url section if you kept it, not the returning url section

Reply:
also i would italicize or bold the last sentence about expiration